executor/wasm/src/lib.rs

```diff
@@ -229,60 +237,104 @@ impl ExecutorV2 {
         // TODO: Michal: why is this result not evaluated?
         let _result = get_purse_for_entity(&mut tracking_copy, caller_key);
 
-        // 1. Store package hash
+        let enable_addressable_entity = tracking_copy.enable_addressable_entity();
         let smart_contract_addr: [u8; 32] = chain_utils::compute_predictable_address(
             chain_name.as_bytes(),
             initiator.value(),
             bytecode_hash,
             seed,
         );
 
-        let mut smart_contract = Package::new(
-            Default::default(),
-            Default::default(),
-            Groups::default(),
-            PackageStatus::Unlocked,
-        );
+        // 1. Store package hash
+        let package_addr: HashAddr = {
```

Comment:
Can we use the deterministic package addresses here? `package_addr` would use `chain_utils::compute_predictable_address` and `smart_contract_addr` would use this https://github.com/casper-network/casper-node/pull/5241/files#diff-90c8dd3c74add596bf2b1097060a4419d3bfa84a3091698d4b04e36785224449 (actually, I see it in the diff later) - I'd prefer deterministic addresses over address generator